recipes-support/host-insight-helper/host-insight-helper/host-insight-helper.sh

```diff
@@ -1,17 +1,47 @@
 #!/bin/sh
 
 ID_PATH="/etc/opt/host-insight-client/identity-fallback.toml"
+CLIENT_UPGRADE="/tmp/host-insight/client_upgrade"
 
-# Get the serial number of the device.
-if [ "$(cat /etc/platform-system-type)" = "c61" ]; then
-  # Sleep until we know that pic_upgrade is complete
-  sleep 30
-  echo "uid = \"$(cat /opt/hm/pic_attributes/ctrl_serial_nr)\"" > $ID_PATH
-else
-  echo "uid = \"$(cat /proc/device-tree/chosen/serial-number)\"" > $ID_PATH
-fi
+upgrade_client()
+{
+  systemctl stop host-insight-client
+  echo "Upgrading host-insight-client to version $new..."
+  read -r new < "$CLIENT_UPGRADE"
+  opkg remove "host-insight-client"
+  if ! opkg install "host-insight-client$new"; then
+    echo "Failed to upgrade to version $new. Falling back to installing 'host-insight-client'"
+    opkg install "host-insight-client"
+  fi
+  rm "$CLIENT_UPGRADE"
+  systemctl start host-insight-client
+}
+
+assign_uid_and_fallback_domain()
```

Comment:
should not assign_uid_and_fallback_domain be runninng after upgrade_client?

Reply:
is it just runing on the first boot after bsp upgrade, if so it will not upgrade the client until the client has been run. but maybe that is the goal design here.